plugins/test/ansible_type.py

```diff
@@ -0,0 +1,203 @@
+result: '{{ {"a": 1} is community.general.ansible_type(dtype) }}'
+# result => true
+
+# Dictionary. All keys are strings. All values are integers.
+dtype: dict[str, int]
+result: '{{ {"a": 1, "b": 2} is community.general.ansible_type(dtype) }}'
+# result => true
+
+# Type of strings is AnsibleUnicode or str
+# ----------------------------------------
+
+# Dictionary. The keys are integers or strings. All values are strings.
+alias: {"AnsibleUnicode": "str"}
+dtype: dict[int|str, str]
+data: {1: 'a', 'b': 'b'}
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+
+# Dictionary. All keys are integers. All values are keys.
+alias: {"AnsibleUnicode": "str"}
+dtype: dict[int, str]
+data: {1: 'a', 2: 'b'}
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+
+# Dictionary. All keys are strings. Multiple types values.
+alias: {"AnsibleUnicode": "str"}
+dtype: dict[str, bool|dict|float|int|list|str]
+data: {'a': 1, 'b': 1.1, 'c': 'abc', 'd': True, 'e': ['x', 'y', 'z'], 'f': {'x': 1, 'y': 2}}
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+
+# List. Multiple types items.
+alias: {"AnsibleUnicode": "str"}
+dtype: list[bool|dict|float|int|list|str]
+data: [1, 2, 1.1, 'abc', True, ['x', 'y', 'z'], {'x': 1, 'y': 2}]
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+
+# Option dtype is list
+# --------------------
+
+# AnsibleUnicode or str
+dtype: ['AnsibleUnicode', 'str']
+data: abc
+result: '{{ data is community.general.ansible_type(dtype) }}'
+# result => true
+
+# float or int
+dtype: ['float', 'int']
+data: 123
+result: '{{ data is community.general.ansible_type(dtype) }}'
+# result => true
+
+# float or int
+dtype: ['float', 'int']
+data: 123.45
+result: '{{ data is community.general.ansible_type(dtype) }}'
+# result => true
+
+# Multiple alias
+# --------------
+
+# int alias number
+alias: {"int": "number", "float": "number"}
+dtype: number
+data: 123
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+
+# float alias number
+alias: {"int": "number", "float": "number"}
+dtype: number
+data: 123.45
+result: '{{ data is community.general.ansible_type(dtype, alias) }}'
+# result => true
+'''
+
+RETURN = '''
+  _value:
+    description: Whether the data type is valid.
+    type: bool
+'''
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common._collections_compat import Sequence
+from ansible_collections.community.general.plugins.plugin_utils.ansible_type import _ansible_type
+
+
+def ansible_type(data, dtype, alias=None):
+    """Validates data type"""
+
+    if not isinstance(dtype, Sequence):
+        msg = "The argument dtype must be a string or a list. dtype is %s."
+        raise AnsibleFilterError(msg % (dtype, type(dtype)))
+
+    if isinstance(dtype, list):
+        data_types = dtype
+    else:
+        data_types = [dtype]
```

Comment:
Tuples are also sequences, so it's better to check explicitly for strings here:
```suggestion
    if isinstance(dtype, str):
        data_types = [dtype]
    else:
        data_types = dtype
```